relay-server/src/services/processor.rs

```diff
@@ -1253,17 +1260,6 @@ impl EnvelopeProcessorService {
                 .send(UpdateRateLimits::new(scoping.project_key, limits));
         }
 
-        if enforcement.event_active() {
-            state.remove_event();
-            debug_assert!(state.envelope().is_empty());
-        }
```

Comment:
Moved up, since it requires enforcement which is consumed by the `apply_with_outcomes`. I verified that the ordering doesn't seem to impact the end result.